diff --git a/lib/trusted-publishing/oidc-context.js b/lib/trusted-publishing/oidc-context.js
index 0427ae9de9da75861b5db715420c9e82a727fbb3..24477c104345c5acf847f107910578a663bad011 100644
--- a/lib/trusted-publishing/oidc-context.js
+++ b/lib/trusted-publishing/oidc-context.js
@@ -2,5 +2,17 @@ import { OFFICIAL_REGISTRY } from "../definitions/constants.js";
 import exchangeToken from "./token-exchange.js";
 
 export default async function oidcContextEstablished(registry, pkg, context) {
-  return OFFICIAL_REGISTRY === registry && !!(await exchangeToken(pkg, context));
+  if (OFFICIAL_REGISTRY !== registry) {
+    context.logger.log(`[@semantic-release/npm]: OIDC registry mismatch: OFFICIAL_REGISTRY="${OFFICIAL_REGISTRY}", registry="${registry}"`);
+    return false;
+  }
+
+  const token = await exchangeToken(pkg, context);
+
+  if (token) {
+    context.env.NPM_TOKEN = token;
+    return true;
+  }
+
+  return false;
 }
diff --git a/lib/verify-auth.js b/lib/verify-auth.js
index d09933760e43b54383b470de4b320eb5a1c1c119..5aa0546db74bfebf3a53438dd9ce4f1bec72b681 100644
--- a/lib/verify-auth.js
+++ b/lib/verify-auth.js
@@ -83,11 +83,15 @@ async function verifyTokenAuth(registry, npmrc, context, pkgRoot) {
 export default async function (npmrc, pkg, { pkgRoot }, context) {
   const registry = getRegistry(pkg, context);
 
-  if (await oidcContextEstablished(registry, pkg, context)) {
-    return;
-  }
+  const isOidc = await oidcContextEstablished(registry, pkg, context);
+  context.logger.log(`[@semantic-release/npm]: isOidc=${isOidc}`);
 
   await setNpmrcAuth(npmrc, registry, context);
 
+  if (isOidc) {
+    context.logger.log(`[@semantic-release/npm]: Skipping verifyTokenAuth because isOidc is true`);
+    return;
+  }
+
   await verifyTokenAuth(registry, npmrc, context, pkgRoot);
 }
